language/stdlib/modules/libra_system.move

```diff
@@ -70,94 +100,255 @@ module LibraSystem {
         });
     }
 
-    // ValidatorInfo public accessors
+    fun add_validator_no_discovery_event(account_address: address) acquires DiscoverySet {
+        Transaction::assert(Transaction::sender() == 0xA550C18, 1);
+        // A prospective validator must have a validator config resource
+        Transaction::assert(ValidatorConfig::has(account_address), 17);
 
-    public fun get_validator_address(v: &ValidatorInfo): &address {
-      &v.addr
-    }
+        let validator_set = get_validator_set();
+        // Ensure that this address is not already a validator
+        Transaction::assert(!is_validator_(account_address, &validator_set.validators), 18);
 
-    public fun get_consensus_pubkey(v: &ValidatorInfo): &vector<u8> {
-      &v.consensus_pubkey
-    }
+        let config = ValidatorConfig::get_config(account_address);
+        Vector::push_back(&mut validator_set.validators, ValidatorInfo {
+            addr: account_address,
+            config: config, // copy the config over to ValidatorSet
+            consensus_voting_power: 1,
+        });
 
-    public fun get_consensus_voting_power(v: &ValidatorInfo): &u64 {
-      &v.consensus_voting_power
+        let discovery_config = ValidatorConfig::get_discovery_config(account_address);
+        let discovery_set_ref = &mut borrow_global_mut<DiscoverySet>(0xD15C0).discovery_set;
+        Vector::push_back(
+            discovery_set_ref, DiscoveryInfo {
+                addr: account_address,
+                discovery_config: discovery_config
+            });
+
+        set_validator_set(validator_set);
     }
 
-    public fun get_network_signing_pubkey(v: &ValidatorInfo): &vector<u8> {
-      &v.network_signing_pubkey
+    // Adds a new validator, only callable by the LibraAssociation address
+    // TODO(valerini): allow the Association to add multiple validators in a single block
+    public fun add_validator(account_address: address) acquires DiscoverySet {
+        add_validator_no_discovery_event(account_address);
+        emit_discovery_set_change();
     }
 
-    public fun get_network_identity_pubkey(v: &ValidatorInfo): &vector<u8> {
-      &v.network_identity_pubkey
+    // Removes a validator, only callable by the LibraAssociation address
+    public fun remove_validator(account_address: address) acquires DiscoverySet {
+        Transaction::assert(Transaction::sender() == 0xA550C18, 1);
+        let validator_set = get_validator_set();
+        let size = Vector::length(&validator_set.validators);
+        // Make sure the size of validator set does not go beyond 4, which is the minimum safe number of validators
+        Transaction::assert(size > 4, 22);
```

Comment:
Should this be `>= 4` given the comment above?

Reply:
It's decremented in the next lines, so to stay >= 4, it should be > 4 here.